### PR TITLE
feat(ctf): CTF workspace SPA (participant + organizer) on typed /api/v1 surface

### LINKED ISSUE #1372: SPA Phase 2: CTF workspace (participant + organizer)
Milestone: SPA Cutover — Phase 2

Blocked by: #1368 (cohesive UX / IA / wireframes), #1369 (app shell & global navigation)

## Context

Port and redesign the CTF GUI (participant + organizer) onto the platform SPA, following the Risk Register pattern (#1302), building against the cohesive design (#1368) inside the shared app shell (#1369).

## Locked foundation (do NOT redesign)

- **Visual theme** — Apple-dark Tailwind v4 + shadcn/ui design system (from #1302).
- **Brand** — Shifter chevron logo + favicon.

Reuse as-is; do not restyle.

## What is open (design freedom)

Layout, information architecture, navigation, what is surfaced where and how, and the flows themselves are open, per #1368. The current CTF UX is not a constraint — rework clunky flows, don't just re-skin. Target a friendlier, more accessible (WCAG 2.1 AA), more usable experience.

## Scope

- Participant: registration, challenge browsing / solving, flag submission, scoreboard / rankings.
- Organizer: event and challenge management, monitoring.

## Constraints

- Data access on the canonical `/api/v1/` DRF surface; no ad hoc endpoints. Where CTF lacks `/api/v1/` coverage, note the gap (API consolidation tracked separately, e.g. #1328 / #1329).
- Preserve auth/session/CSRF and existing CTF privacy semantics (e.g. solve-history visibility is own-participant + organizer only); do not weaken them.
- Keep legacy Django routes until a rollout plan says removal is safe; ship behind a rollout flag.
- Reuse the #1302 design system + logo; no new visual language.

## Acceptance criteria

- Core CTF workflows (register, browse/solve challenges, submit flags, view scoreboard; organizer manage/monitor) work in the SPA behind a rollout flag, with the legacy path preserved for rollback.
- Loading / empty / validation / permission / error / destructive-confirm states handled.
- Accessibility AA.
- Applicable frontend, Django, and ADR guardrail checks pass for touched files.

### PR DESCRIPTION
**Pull request by Brad-Edwards**
## Summary

Port the CTF GUI (participant + organizer) onto the platform SPA (#1372), behind a default-off CTF_WORKSPACE_SPA_ENABLED flag AND-gated with PLATFORM_SPA_ENABLED, with the legacy Django pages preserved as the default + rollback path. Two halves on one branch: (1) a full DRF rewrite that gives CTF a typed, published /api/v1/ctf/ OpenAPI surface (the blockers #1328/#1329 deliberately deferred CTF typing into this issue — verified via the exclusion hook + design note); (2) the participant + organizer React workspace built on the existing feature-flagged shell, consuming the generated types (no hand-typed DTOs, per the binding preflight). Data access is on /api/v1/ only; auth/session/CSRF and CTF privacy semantics (participant-safe projections filter flags/solutions/validator-config server-side; solve-history own-participant + organizer only) are preserved; registration/magic-link stays server-owned.

## Requirement UIDs

- (none — bug/refactor/maintenance run; see Traceability section below)

## Related Issues

Closes #1372

## ADR Impact

- ADR-013
- ADR-029

## Changes

- Rewrite every /api/v1/ctf/ endpoint from transitional legacy_api_view wrappers into proper DRF views (request serializer -> ctf.services.* -> typed response serializer); retire the CTFLegacyAPIView wrapper machinery entirely
- Add net-new participant read projections the legacy template views never exposed as JSON: current-event, challenge browse, challenge detail-for-play, team (ctf/api/participant_views.py + projections.py + shared ctf/services/participant/play.py)
- Un-exclude ctf. from the published OpenAPI contract and regenerate openapi/v1.json + frontend/src/api/schema.d.ts (45 typed CTF paths); contract drift + breaking-change gates pass (additive)
- Build the participant workspace (event home, challenges browse/solve, flag submission, scoreboard, team, range) and organizer workspace (event/challenge/participant management, monitoring) under frontend/src/features/ctf on the existing shell
- Rollout plumbing: CTF_WORKSPACE_SPA_ENABLED flag (settings + env-manifest + bootstrap), a method-aware ctf/urls.py SPA host (shell for safe methods when on, legacy Django otherwise), nav entries flipped internal behind the flag, ctf route groups gated on ctf_participant/ctf_organizer
- Tests: per-endpoint DRF contract preservation (existing suites unchanged), participant-safety schema assertions, SPA-host flag behavior, and participant + organizer page tests

## Test Plan

- [x] Unit tests pass (`make test`)
- [x] Integration tests pass if applicable (`make integration`)
- [x] `make check` passes (Spotless, SpotBugs, Error Prone, Checkstyle, JaCoCo)
- [x] No coverage regression

Backend full pytest suite green on every commit via pre-commit (non-postgres/non-redis gate). Frontend: tsc --noEmit clean, eslint clean, vitest 250/250. OpenAPI contract --check clean (no drift); CTF participant types never expose flag_hash/flag_format. Concurrency tests are postgres-marked (excluded from the sqlite gate, run on CI's postgres lane).

## Ground Control Checks

- [x] `make policy` passes
- [x] `gc_evaluate_quality_gates` passes or is unchanged by this repo-only change
- [x] `gc_run_sweep` reviewed; findings fixed or recorded with rationale

## Traceability

- IMPLEMENTS: ADR-013, ADR-029
- TESTS: shifter/shifter_platform/tests/ctf/test_participant_api.py, shifter/shifter_platform/tests/ctf/test_ctf_api_contract.py, shifter/shifter_platform/tests/ctf/test_spa_host.py

## Checklist

- [x] Code follows project coding standards (`docs/CODING_STANDARDS.md`)
- [x] No business logic in API layer
- [x] Domain layer has no framework imports
- [x] Envers `@Audited` on new entities if applicable
- [x] Changelog fragment added at `changelog.d/1372.added.md`
- [x] Architectural docs updated if stack, package structure, or key behaviors changed

### CODE REVIEW

File: shifter/shifter_platform/frontend/src/features/ctf/admin/ChallengeFormPage.tsx [hunk 1]
```diff
@@ -0,0 +1,667 @@
...
+    name: challenge.name ?? "",
+    description: challenge.description ?? "",
+    category: challenge.category || "web",
+    difficulty: challenge.difficulty || "easy",
+    points: String(challenge.points ?? 0),
+    order: String(challenge.order ?? 0),
+    max_attempts: String(challenge.max_attempts ?? 0),
+    flag_format: challenge.flag_format ?? "",
+    solution: challenge.solution ?? "",
+    visibility: "visible",
```
Comment (Brad-Edwards) [inline shifter/shifter_platform/frontend/src/features/ctf/admin/ChallengeFormPage.tsx]: [core] Incomplete detail contracts cause organizer controls to substitute false state

_class finding — category: Organizer management controls substitute hard-coded defaults for persisted state omitted from their detail DTOs. — instances: shifter/shifter_platform/frontend/src/features/ctf/admin/ChallengeFormPage.tsx:89, shifter/shifter_platform/frontend/src/features/ctf/admin/ParticipantDetailPage.tsx:123. Fix the category, not just this site._

OrganizerChallengeDetail omits visibility, target_instance_name, and target_port, so fromChallenge substitutes "visible" and empty target values; toPayload then sends those values on every PUT. Editing an existing hidden/locked or range-targeted challenge therefore silently makes it visible and clears its target configuration. The same contract-shape problem appears in ParticipantDetailPage: the participant detail omits the current bracket and the control always initializes to "No bracket." Extend the organizer detail projections with the persisted values required by these management controls, hydrate them rather than inventing defaults, and pin non-default edit round trips in tests.

File: shifter/shifter_platform/frontend/src/api/ctf.ts [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+    queryKey: ctfKeys.rangeStatus(),
+    queryFn: ({ signal }) => apiFetch<CtfRangeStatus>(`${BASE}/range/status/`, { signal }),
+  });
+}
+
+export function useCtfScoreboard(eventId: string, bracketId?: string, enabled = true) {
+  return useQuery({
+    queryKey: ctfKeys.scoreboard(eventId, bracketId),
+    enabled: enabled && Boolean(eventId),
+    queryFn: ({ signal }) =>
```
Comment (Brad-Edwards) [inline shifter/shifter_platform/frontend/src/api/ctf.ts:133]: [core] Organizer monitoring uses the public scoreboard policy

The shared hook always calls the route backed by PublicScoreboardView. That view is AllowAny, hides the board when scoreboard_visible is false, and applies the freeze cutoff to every caller. MonitoringPage uses this hook for the organizer workspace, so organizers can receive an empty/hidden or stale frozen board instead of the live operational view. The incumbent authenticated scoreboard explicitly gives an owning organizer unfrozen access regardless of participant visibility. Publish a typed role-aware organizer scoreboard contract, or split the participant and organizer hooks so each calls the correct policy, and test organizer behavior for hidden and frozen events.

**Comment by Brad-Edwards** [comment]
_gc_codex_review cycle 1 of 3 complete for PR #1689._ Posted by the MCP server to enforce the hard-cap-3 contract (issues #794, #804). Do not edit or delete — used by the next `gc_codex_review` invocation to count cycles.

**Comment by Brad-Edwards** [comment]
_gc_codex_verify_finding cycle 1 of 2 complete for PR #1689 comment #3592119337._ Posted by the MCP server to enforce the per-finding hard-cap-2 contract (issue #794). Do not edit or delete — used by the next `gc_codex_verify_finding` invocation to count cycles.

File: shifter/shifter_platform/frontend/src/features/ctf/admin/ChallengeFormPage.tsx (hunk 1 above)
Comment (Brad-Edwards) [reply shifter/shifter_platform/frontend/src/features/ctf/admin/ChallengeFormPage.tsx]: The production hydration is corrected, but the required non-default regression coverage is still missing. In `ChallengeFormPage.test.tsx`, add an edit test whose detail response has `visibility: "hidden"` (or `"locked"`), a non-empty `target_instance_name`, and a non-null `target_port`; submit “Save changes” and assert the PUT payload preserves all three values. In `ParticipantDetailPage.test.tsx`, provide a non-null `bracket_id` plus matching organizer-scoreboard bracket data and assert `BracketControl` initially selects that bracket rather than “No bracket.” The current tests omit these fields and therefore would not catch this regression.

File: shifter/shifter_platform/frontend/src/api/ctf.ts (hunk 2 above)
Comment (Brad-Edwards) [reply shifter/shifter_platform/frontend/src/api/ctf.ts:133]: The route split and typed organizer contract are implemented, but the required regression coverage is missing. Add endpoint tests for `OrganizerScoreboardView` in `shifter/shifter_platform/tests/ctf/`: an owning organizer must receive rankings when `scoreboard_visible=False`, and a frozen event must include post-freeze scoring data by exercising `freeze_at=None`. The existing frozen-organizer test covers participant solve-history HTML, not `/api/v1/ctf/events/{event_id}/organizer-scoreboard/`. Also make `MonitoringPage.test.tsx` assert that monitoring requests `/ctf/events/e1/organizer-scoreboard/`, so reverting to the public hook fails.

**Comment by Brad-Edwards** [comment]
_gc_codex_verify_finding cycle 1 of 2 complete for PR #1689 comment #3592119392._ Posted by the MCP server to enforce the per-finding hard-cap-2 contract (issue #794). Do not edit or delete — used by the next `gc_codex_verify_finding` invocation to count cycles.

**Comment by Brad-Edwards** [comment]
_gc_codex_verify_finding cycle 2 of 2 complete for PR #1689 comment #3592119337._ Posted by the MCP server to enforce the per-finding hard-cap-2 contract (issue #794). Do not edit or delete — used by the next `gc_codex_verify_finding` invocation to count cycles.

**Comment by Brad-Edwards** [comment]
_gc_codex_verify_finding cycle 2 of 2 complete for PR #1689 comment #3592119392._ Posted by the MCP server to enforce the per-finding hard-cap-2 contract (issue #794). Do not edit or delete — used by the next `gc_codex_verify_finding` invocation to count cycles.
